source/common/config/context_provider_impl.h

```diff
@@ -13,10 +14,35 @@ class ContextProviderImpl : public ContextProvider {
                       const Protobuf::RepeatedPtrField<std::string>& node_context_params)
       : node_context_(XdsContextParams::encodeNodeContext(node, node_context_params)) {}
 
+  // Config::ContextProvider
   const xds::core::v3::ContextParams& nodeContext() const override { return node_context_; }
+  const xds::core::v3::ContextParams&
+  dynamicContext(absl::string_view resource_type_url) const override {
+    auto it = dynamic_context_.find(resource_type_url);
+    if (it != dynamic_context_.end()) {
+      return it->second;
+    }
+    return xds::core::v3::ContextParams::default_instance();
+  };
+  void setDynamicContextParam(absl::string_view resource_type_url, absl::string_view key,
```

Comment:
I think it'd be nice if this could be a post CB on the main thread, otherwise there is a racing problem between callers from another thread and the readers of the context parameters in main thread.

Reply:
Alternatively maybe we can just state a requirement (or maybe there already is one?) that ContextProvider only gets called from the main thread and leave callback posting to the outside code?

Reply:
the race is between callers of this setDCP method and the main thread grpc mux. Unless there is a lock,  otherwise we need to use the Post-main-dispatcher-CB trick to avoid racing.

Reply:
All these callbacks for config are happening on the main thread, there should be no race.

Reply:
We have new main thread assertions now. Can we use them?